arangod/Scheduler/Scheduler.cpp

```diff
@@ -370,16 +446,26 @@ bool Scheduler::pushToFifo(size_t fifo, std::function<void()> const& callback) {
   return true;
 }
 
-bool Scheduler::popFifo(size_t fifo) {
-  int64_t p = fifo - 1;
-  TRI_ASSERT(0 <= p && p < NUMBER_FIFOS);
+bool Scheduler::popFifo(int64_t fifo) {
+  TRI_ASSERT(0 <= fifo && fifo < NUMBER_FIFOS);
+
+  if (fifo == FIFO8 && _queuedV8 >= _maxQueuedV8) {
+    return false;
+  }
 
+  size_t p = static_cast<size_t>(fifo);
   FifoJob* job = nullptr;
+
   bool ok = _fifos[p]->pop(job) && job != nullptr;
 
   if (ok) {
-    post(job->_callback);
-    delete job;
+    if (!job->_isV8 || _queuedV8 < _maxQueuedV8) {
+      post(job->_callback, job->_isV8);
```

Comment:
we are posting the job here, and instantly delete it afterwards.
Isn't this safe? does anyone in the callback access `job`?

Reply:
Apart from that, what happens if `post` throws? Then we wil leak `job`.

Reply:
done